Userland/Libraries/LibCompress/Zlib.cpp

```diff
@@ -68,21 +69,22 @@ u32 ZlibDecompressor::checksum()
     return m_checksum;
 }
 
-ErrorOr<NonnullOwnPtr<ZlibCompressor>> ZlibCompressor::construct(OutputStream& stream, ZlibCompressionLevel compression_level)
+ErrorOr<NonnullOwnPtr<ZlibCompressor>> ZlibCompressor::construct(Core::Stream::Handle<Core::Stream::Stream> stream, ZlibCompressionLevel compression_level)
 {
     // Zlib only defines Deflate as a compression method.
     auto compression_method = ZlibCompressionMethod::Deflate;
 
-    auto zlib_compressor = TRY(adopt_nonnull_own_or_enomem(new (nothrow) ZlibCompressor(stream, compression_level)));
-    zlib_compressor->write_header(compression_method, compression_level);
+    auto zlib_compressor = TRY(adopt_nonnull_own_or_enomem(new (nothrow) ZlibCompressor(move(stream), compression_level)));
+    TRY(zlib_compressor->write_header(compression_method, compression_level));
 
     return zlib_compressor;
 }
 
-ZlibCompressor::ZlibCompressor(OutputStream& stream, ZlibCompressionLevel compression_level)
-    : m_output_stream(stream)
+ZlibCompressor::ZlibCompressor(Core::Stream::Handle<Core::Stream::Stream> stream, ZlibCompressionLevel compression_level)
+    : m_ak_output_stream(make<Core::Stream::WrapInAKOutputStream>(*stream))
```

Comment:
Again, construct this outside and then pass it in.

Reply:
Kind of the same argument as with `DeflateCompressor` above (as it is fixed in cad2137b2c0cf09a6b91ba350a6eef8a01433bf2 in the next PR, after which we no longer need the wrapper here), but I guess this one _is_ new for the time being, so I'll look into it if you deem it necessary (I'd just like to avoid creating conflicts for myself where possible).